### ISSUE TITLE
NilClass errors when package info fails

### ISSUE THREAD

**Issue by adamleff** [inspec/inspec#1692]
Appears to be a bug in the `package` resource if the command used to find the package details fails.

```
              1) System Package amazon-ssm-agent should be installed
                   Failure/Error: info[:installed] == true

                        NoMethodError:
                               undefined method `[]' for nil:NilClass
```

`info` should return an empty hash until we have better exception handling.

**Comment by jerryaldrichiii** [comment]
@adamleff, is this still an issue? I see that we return `false` if `info.nil?`

https://github.com/chef/inspec/blob/master/lib/resources/package.rb#L56

**Comment by jerryaldrichiii** [comment]
I like the idea of returning `{}` though. I think it's cleaner.

Could you expand on your thoughts about better exception handling now that we have the `ResourceSkipped`/`ResourceFailed` exceptions?

**Comment by jerryaldrichiii** [comment]
In the meantime, here's a quick stab at having `info` return `{}` and some exceptions being tweaked.

Let me know what you think.

https://gist.github.com/jerryaldrichiii/ea4be62ff4c0542d8d5f0b40355a4a41

**Comment by jerryaldrichiii** [comment]
This also fixes #2373  😄
